Frontend/src/app/insert-match/insert-match.component.ts

```diff
@@ -61,6 +61,12 @@ export class InsertMatchComponent implements OnInit, OnChanges {
 			this.error = {description: 'Verifique se há um campeonato criado.'};
 			return false;
 		}
+
+		if (this.matchService.getFinals() && this.match.isFinal) {
```

Comment:
I did, I just thought empty arrays would be false if checked as booleans.. Just a small change and it's ready to go.